comment/api/views.py

```diff
@@ -131,15 +133,46 @@ def post(self, request, *args, **kwargs):
 
 
 class ConfirmComment(APIView):
-    @staticmethod
-    def get(request, *args, **kwargs):
+    email_service = None
+
+    def get(self, request, *args, **kwargs):
         key = kwargs.get('key', None)
-        comment = get_comment_from_key(key)
+        temp_comment = get_comment_from_key(key)
 
-        if comment.why_invalid == CommentFailReason.BAD:
+        if temp_comment.why_invalid == CommentFailReason.BAD:
             return Response({'detail': EmailError.BROKEN_VERIFICATION_LINK}, status=status.HTTP_400_BAD_REQUEST)
 
-        if comment.why_invalid == CommentFailReason.EXISTS:
+        if temp_comment.why_invalid == CommentFailReason.EXISTS:
             return Response({'detail': EmailError.USED_VERIFICATION_LINK}, status=status.HTTP_200_OK)
 
-        return Response(CommentSerializer(comment.obj).data, status=status.HTTP_201_CREATED)
+        comment = temp_comment.obj
+        comment.save()
+        comment.refresh_from_db()
+        if settings.COMMENT_ALLOW_SUBSCRIPTION:
+            self.email_service = DABEmailService(comment, request)
+            self.email_service.send_notification_to_followers()
```

Comment:
Maybe instead of doing this here inside the `views`, we can probably register this function to a signal that uses the email service to send messages directly when we parent comments are created.

Reply:
I thought about this in the first place but the issue is that initializing email service needs the request object and it is not available within the signals. I believe there is workaround for this issue but it is not worth for our case.
I do agree with django that the request object shall be used from within the view only.

Reply:
This just means that any possible changes to email service will now have to be done possibly at more than one place because of code duplication in our case. Something to consider maybe.